feature_engine/selection/single_feature_performance_selection.py

```diff
@@ -0,0 +1,171 @@
+        sklearn.metrics. See the model evaluation documentation for more options:
+        https://scikit-learn.org/stable/modules/model_evaluation.html
+
+    threshold: float, int, default = 0.01
+        The value that defines if a feature will be kept or removed. Note that for
+        metrics like roc-auc, r2_score and accuracy, the thresholds will be floats
+        between 0 and 1. For metrics like the mean_square_error and the
+        root_mean_square_error the threshold will be a big number.
+        The threshold must be defined by the user.
+
+    cv : int, default=3
+        Desired number of cross-validation fold to be used to fit the estimator.
+
+    Attributes
+    ----------
+
+    initial_model_performance_: float,
+        performance of the model built using the original dataset.
+
+    selected_features_: list
+        The selected features.
+
+    Methods
+    -------
+
+    fit: finds important features
+
+    transform: removes non-important / non-selected features
+
+    fit_transform: finds and removes non-important features
+
+    """
+
+    def __init__(
+        self,
+        estimator=RandomForestClassifier(),
+        scoring="roc_auc",
+        cv=3,
+        threshold=0.5,
+        variables=None,
+    ):
+
+        if not isinstance(cv, int) or cv < 1:
+            raise ValueError("cv can only take positive integers bigger than 1")
+
+        if not isinstance(threshold, (int, float)):
+            raise ValueError("threshold can only be integer or float")
+
+        self.variables = _define_variables(variables)
+        self.estimator = estimator
+        self.scoring = scoring
+        self.threshold = threshold
+        self.cv = cv
+
+    def fit(self, X, y):
+        """
+
+        Args
+        ----
+
+        X: pandas dataframe of shape = [n_samples, n_features]
+           The input dataframe
+
+        y: array-like of shape (n_samples)
+           Target variable. Required to train the estimator.
+
+
+        Returns
+        -------
+
+        self
+        """
+
+        # check input dataframe
+        X = _is_dataframe(X)
+
+        # find numerical variables or check variables entered by user
+        self.variables = _find_numerical_variables(X, self.variables)
+
+        # list to collect selected features
+        self.selected_features_ = []
+
+        self.feature_importance_ = {}
+
+        # train model with all features and cross-validation
+        for feature in self.variables:
+            model = cross_validate(
+                self.estimator,
+                X[feature].to_frame(),
+                y,
+                cv=self.cv,
+                return_estimator=False,
+                scoring=self.scoring,
+            )
+
+            if model["test_score"].mean() > self.threshold:
+                self.selected_features_.append(feature)
+                self.feature_importance_[feature] = model["test_score"].mean()
```

Comment:
this line should be outside the if, because we want to store the importance for every feature, whether they are selected or not. So that the user can have a look at why a feature may have been selected.